### PR TITLE
Feat: Support ciphers SSLOptions

### PR DESCRIPTION
**Pull request by saiqulhaq**
To resolve one of these [tasks](https://github.com/lostisland/faraday/issues/1586)

### CODE REVIEW

**Comment by saiqulhaq** [comment]
image
CI are success

**Comment by saiqulhaq** [comment]
image
```
Finished in 0.45295 seconds (files took 0.179 seconds to load)
353 examples, 0 failures
```

Faraday 2.8 Ruby 2.7
```
Finished in 0.40026 seconds (files took 0.16809 seconds to load)
353 examples, 0 failures
```

**Review by iMacTia — APPROVED** [review]
Left a couple of minor comments on the versioning, but I can do those changes quite quickly myself.
Apologies if it took this long to review this, and thank you for going through the various adapters to add support for `ciphers` 🙇

File: Gemfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ source "https://rubygems.org"
 
 gemspec
 
-gem "faraday", ">= 1"
+gem "faraday", ">= 2.5", "< 3"
```
[the remark this answers is not in the document]
Comment (saiqulhaq) [reply Gemfile:7]: do you mean just need to use `gem "faraday"`?

[the remark this answers is not in the document]
Comment (iMacTia) [reply Gemfile:7]: No I mean this line can go, it shouldn't be needed at all!

File: lib/faraday/adapter/net_http_persistent.rb [hunk 2]
```diff
@@ -185,7 +185,13 @@ def request_with_wrapped_block(http, env)
         min_version: :min_version,
         max_version: :max_version,
         verify_hostname: :verify_hostname
-      }.freeze
+      }
+
+      if Gem::Version.new(Faraday::VERSION) >= Gem::Version.new('2.11.0')
```
Comment (iMacTia) [inline lib/faraday/adapter/net_http_persistent.rb:190]:
```suggestion
      if Gem::Version.new(Faraday::VERSION) >= Gem::Version.new("2.11.0")
```

I forgot we use `standard.rb` for code style in this repo, it's complaining about the use of single quotes here. Could you change it to double quotes?

File: Gemfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (iMacTia) [reply Gemfile:7]:
```suggestion
```

Running some tests with PR suggestions, let's see if this works

**Comment by iMacTia** [comment]
Managed to do those small changes myself with the web editor (pretty cool!).
All tests seem to be passing now, thank you @saiqulhaq !

**Pull request merged**